lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2404,10 +2403,19 @@ class SqlParser(private val ion: IonSystem) : Parser {
                 }
             }
         }
+        // Extract the precision from the time if the precision is not specified.
+        // For e.g., TIME '23:12:12.123' should have precision of 3.
+        // Note that TIME '23:12:12.1230' will also have precision of 3 unless specified otherwise as the significant
+        // digits in the fractional part excluding trailing zeros is 3.
```

Comment:
I would not expect `.1230` to have a precision of 3, but rather a precision of 4, what am I missing?

Reply:
Changed the code to infer precision based on the length of the mantissa.